draco/core/containers.py

```diff
@@ -3340,3 +3340,227 @@ def copy_datasets_filter(
+            "initialise": True,
+            "distributed": False,
+        },
+        "window": {
+            "axes": ["pol", "ew", "freq", "freq"],
+            "dtype": np.complex128,
+            "initialise": True,
+            "distributed": False,
+        },
+    }
+
+    #_data_dset_name = "stack"                                                                                                                                                                                                                                                                                                                                              
+    #_weight_dset_name = "weight"                                                                                                                                                                                                                                                                                                                                           
+
+    @property
+    def bandpass(self):
+        """Get the bandpass dataset."""
+        return self.datasets["bandpass"]
+
+    @property
+    def window(self):
+        """Get the window dataset."""
+        return self.datasets["window"]
+
+
+class VisBandpassCompensateBaseline(FreqContainer):
+    """Container for window-compensated bandpass gains"""
+
+    _axes = ("pol", "ew", "freq")
+
+    _dataset_spec: ClassVar = {
+        "comp_bandpass": {
+            "axes": ["pol", "ew", "freq"],
+            "dtype": np.complex128,
+            "initialise": True,
+            "distributed": False,
+        },
+        "sval": {
+            "axes": ["pol", "ew", "freq"], # want an axis that has the same length as the freq axis "mode"                                                                                                                                                                                                                                                                        
+            "dtype": np.complex128,
+            "initialise": True,
+            "distributed": False,
+        },
+    }
+
+    @property
+    def comp_bandpass(self):
+        """Get the comp_bandpass dataset."""
+        return self.datasets["comp_bandpass"]
+
+    @property
+    def sval(self):
+        """Get the sval dataset."""
+        return self.datasets["sval"]
+
+    
+class VisBandpassWindowBaselineRA(FreqContainer):
+    """Container for bandpass gains and their window estimated by running bandpass HyFoReS on hybrid beam-formed visibilities"""
+
+    _axes = ("pol", "ew", "ra", "freq") #Q8: do I need to declare _axes? What is the point of this? might have to add pol                                                                                                                                                                                                                                                               
```

Comment:
For data with an ra axis, you should inherit from `SiderealContainer`. In this case, you can inherit from both `VisBandpassWindowBaseline` and `SiderealContainer`, and then you do not have to define `_axes` and you _only_ have to add the `_dataset_spec`